components/TokenManagePage/ImageBox.tsx

```diff
@@ -0,0 +1,59 @@
+import Spinner from "../ui/spinner";
+import {getFetchableUrl} from "@/lib/protocolSdk/ipfs/gateway";
+import Image from "next/image";
+import React, {Fragment} from "react";
+import {useZoraManageProvider} from "@/providers/ZoraManageProvider";
+import EditButton from "@/components/TokenManagePage/EditButton";
+import {useTokenProvider} from "@/providers/TokenProvider";
+
+interface ImageBoxProps {
+    handleImageClick: () => void;
+}
+const Container = ({
+                       children,
+                       className = "",
+                   }: {
+    children: React.ReactNode;
+    className?: string;
+}) => (
+    <div className={`size-full flex justify-center ${className}`}>{children}</div>
+);
+
+const ImageBox = ({ handleImageClick }: ImageBoxProps) => {
+    const {isOwner} = useTokenProvider();
+    const {
+        fileUploading,
+        imageUri,
+        pctComplete,
+        previewSrc,
+    } = useZoraManageProvider();
+
+    if (fileUploading) {
+        return (
+            <Container className="flex flex-col items-center gap-2">
+                <Spinner />
+                <p className="font-archivo text-xl">{pctComplete} %</p>
+            </Container>
```

Comment:
Single Responsibility Principle
- Actual: Loader is defined within the ImageBox component file.
- Required: Define a new component file for `components/TokenManagePage/UploadSpinner.tsx` and import the component here in `ImageBox`.